File: AppGenerateFiles/facturacion/Model/Tbl_Compra.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Tbl_Compra : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_Compra { get; set; }
       public string? DatosCompra { get; set; }
       public int? Id_Proveedor { get; set; }
       public DateTime? Fecha { get; set; }
       public Double? Tasa_Cambio { get; set; }
       public string? Moneda { get; set; }
       public Double? Sub_Total { get; set; }
       public Double? Iva { get; set; }
       public Double? Total { get; set; }
       public string? Estado { get; set; }
       [ManyToOne(TableName = "Cat_Proveedor", KeyColumn = "Id_Proveedor", ForeignKeyColumn = "Id_Proveedor")]
       public Cat_Proveedor? Cat_Proveedor { get; set; }
       [OneToMany(TableName = "Detalle_Compra", KeyColumn = "Id_Compra", ForeignKeyColumn = "Id_Compra")]
       public List<Detalle_Compra>? Detalle_Compra { get; set; }
   }
}

File: UI/wwwroot/Facturacion/FrontModel/Tbl_Compra.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { Cat_Proveedor } from './Cat_Proveedor.js'
import { Detalle_Compra } from './Detalle_Compra.js'
class Tbl_Compra extends EntityClass {
    constructor(props) {
        super(props, 'EntityFacturacion');
        for (const prop in props) {
            this[prop] = props[prop];
        }
    }
   /**@type {Number}*/ Id_Compra;
   /**@type {String}*/ Datos_Compra;
   /**@type {Date}*/ Fecha;
   /**@type {Number}*/ Tasa_Cambio;
   /**@type {String}*/ Moneda;
   /**@type {Number}*/ Sub_Total;
   /**@type {Number}*/ Iva;
   /**@type {Number}*/ Total;
   /**@type {String}*/ Estado;
   /**@type {Cat_Proveedor} ManyToOne*/ Cat_Proveedor;
   /**@type {Array<Detalle_Compra>} OneToMany*/ Detalle_Compra;
    Anular = async () => {
        return await this.SaveData("apiEntityFacturacion/anularCompra",this);
    }
}
export { Tbl_Compra }
